tests/lib/enableRendererCoverage.js

```diff
@@ -0,0 +1,273 @@
+ *       to the running coverage server which stores this BrowserWindow coverage information;
+ *     - Note that single BrowserWindows type (e.g. QSS) would get multiple coverage data
+ *       reports (for every test sequence) that might test different elements of the renderer code. These coverage files
+ *       are then merged by "istanbul" (nyc) in the final coverage report.
+ */
+
+
+
+/**
+ * Extension of the `gpii.app.dialog` that is to be used for coverage collecting.
+ */
+fluid.defaults("gpii.tests.app.instrumentedDialog", {
+    // ensure that `gpii.app.dialog`s are using the instrumented version of renderer content.
+    config: {
+        filePrefixPath: "/instrumented/src/renderer"
+    },
+    listeners: {
+        "onCreate.registerDomStateListener": {
+            funcName: "gpii.tests.app.domStateListener",
+            args: ["{that}.dialog"]
+        },
+        "onCreate.disableWindowDestr": {
+            func: "gpii.tests.app.instrumentedDialog.disableWindowDestr",
+            args: "{that}"
+        },
+        "onDestroy.cleanupElectron": {
+            func: "gpii.tests.app.instrumentedDialog.disableCleanUpDestr",
+            args: "{that}"
+        }
+    }
+});
+
+/**
+ * Attach flag to that BrowserWindow object that specify whether the DOM of the
+ * window is fully initialized. This is needed as `webContents.executeJavaScript`
+ * can only be used in case the DOM is ready.
+ * @param {Component} dialog - The BrowserWindows instance
+ */
+gpii.tests.app.domStateListener = function (dialog) {
+    dialog.domReady = false;
+
+    dialog.webContents.on("dom-ready", function () {
+        dialog.domReady = true;
+    });
+};
+
+/**
+ * Disable destruction of the BrowserWindow of `closable` components as we need to collect their
+ * coverage first. They should be the only type of dialogs that are being destroyed throughout a gpii-app run.
+ * @param {Component} that - The `gpii.app.dialog` instance
+ */
+gpii.tests.app.instrumentedDialog.disableCleanUpDestr = function (that) {
+    if (!that.dialog.isDestroyed()) {
+        if (that.options.config.closable) {
+            /*
+             * Keep information about the dialog's wrapping component. This is
+             * useful when dialog's destruction is postponed e.g. when coverage
+             * collecting is needed.
+             */
+            that.dialog.componentDestroyed = true;
+
+            // disable destruction
+            that.dialog.hide();
+        } else {
+            that.dialog.destroy();
+        }
+    }
+};
+
+/**
+ * This method is only applicable to dialogs that are destroyed when closed and
+ * prevents destruction of the dialog in order for its coverage data to be collected
+ * @param {Component} that - The `gpii.app.dialog` instance
+ */
+gpii.tests.app.instrumentedDialog.disableWindowDestr = function (that) {
+    if (that.options.config.closable) {
+        var dialog = that.dialog;
+
+        dialog.on("close", function (e) {
```

Comment:
What if there is already a handler for the "close" event which is defined in the wrapping component?

Reply:
It should be overridden